src/app/pages/main/main.component.ts

```diff
@@ -387,20 +411,45 @@ export class MainComponent implements OnInit {
    * @param menuItem string identificando qual menu foi clicado
    * @returns {void} - Não retorna nenhum valor.
    */
-  onMenuSelect(menuItem: string) {
+  public onMenuSelect(menuItem: string) {
     this.selectedMenu = menuItem as SelectedMenu;
     this.closeMenus();
   }
 
   /**
-   * Verifica se deve mostrar o card manager
-   *
-   * Este método avalia se o menu selecionado é o de "aguardando voluntário".
-   *
-   * @returns {boolean} - Se a condição for verdadeira, o componente LessonCardManager será exibido.
+   * Mostra um componente dependendo do tipo de usuário e qual botão foi selecionado
+   * @param componentType O tipo do componente a ser verificado
+   * @returns Retorna 'true' se o componente correspondente deve ser exibido, 'false' caso contrário
    */
-  shouldShowCardManager(): boolean {
-    return this.selectedMenu === SelectedMenu.WAITING_VOLUNTEER;
+  public shouldShowComponent(componentType: string): boolean {
```

Comment:
A gente pode melhorar a legibilidade desse código de muitas formas:

### **1. Remover duplicação de lógica**
A lógica `this.selectedMenu === ... && !this.showProfile` se repete várias vezes. Isso pode ser abstraído em um método auxiliar para reduzir duplicação e centralizar a lógica.

Reply:
### **2. Usar mapas para associar tipos de usuários e menus**
Em vez de usar vários `switch` cases, um mapa (ou objeto) pode associar o tipo de componente ao menu correto com base no tipo de usuário. Isso simplifica a lógica e facilita alterações futuras.

Reply:
### **3. Evitar encadeamento de `if/else`**
O uso de `if/else` encadeados pode ser substituído por uma abordagem mais declarativa com estruturas como objetos, melhorando a legibilidade e reduzindo a complexidade.

Reply:
### **Exemplo Refatorado**

```typescript
public shouldShowComponent(componentType: string): boolean {
  const studentMenus = {
    begin: SelectedMenu.BEGIN,
    first: SelectedMenu.WAITING_VOLUNTEER,
    second: SelectedMenu.WAITING_CONFIRMATION,
    third: SelectedMenu.CLASSES_SCHEDULED,
  };

  const tutorMenus = {
    begin: SelectedMenu.TUTOR_BEGIN,
    first: SelectedMenu.TUTOR_SCHEDULED,
    second: SelectedMenu.TUTOR_REQUEST,
    third: SelectedMenu.TUTOR_WAITING_CONFIRMATION,
  };

  const userMenus = this.isStudent() ? studentMenus : this.isTutor() ? tutorMenus : null;

  if (!userMenus) return false;

  const selectedMenu = userMenus[componentType];
  return selectedMenu === this.selectedMenu && !this.showProfile;
}
```

Reply:
Com melhorias como essa, já dá pra perceber como o código fica mais claro e fácil de manter. Centralizar a lógica de associação entre tipos de componentes e menus em um único lugar simplifica bastante a expansão futura, como adicionar novos tipos de usuários ou componentes e a gente acaba eliminando as repetições, o método fica mais direto e seu propósito bem mais evidente, o que ajuda muito na hora de revisar ou fazer ajustes por exemplo :)

OBS: Reflita nas decisões, exemplo: se validações como essa forem comuns na aplicação, dá pra pensar em levar isso pra um serviço centralizado. Usar algo como o padrão **Strategy** seria uma boa ideia, porque encapsularia as regras em estratégias reutilizáveis, deixando tudo mais organizado e ainda mais fácil de testar. Dá até pra escalar essa abordagem em outras partes do sistema, o que só melhora a consistência e a manutenção. 
O *Refactoring Guru* explica bem o **Strategy** e outros padrões de [Design-Patterns](https://refactoring.guru/design-patterns)